eldritch/expansions/seaside/abilities.py

```diff
@@ -0,0 +1,114 @@
+import typing
+
+from eldritch import assets
+from eldritch import characters
+from eldritch import events
+from eldritch import values
+
+if typing.TYPE_CHECKING:
+  from eldritch.eldritch import GameState
+
+
+class AbnormalFocus(assets.Asset):
+  def __init__(self):
+    super().__init__("Abnormal Focus")
+
+  def get_interrupt(self, event, owner: characters.BaseCharacter, state):
+    if isinstance(event, events.Upkeep) and event.character == owner:
+      return events.MoveSliders(owner, {slider + "_slider": 0 for slider in owner.sliders()})
+    return None
+
+
+class BreakingTheLimits(assets.Asset):
+  def __init__(self):
+    super().__init__("Breaking the Limits")
+    self.tokens["sanity"] = 0
+    self.tokens["stamina"] = 0
+
+  def get_interrupt(self, event, owner, state):
+    if isinstance(event, events.Upkeep) and event.character == owner:
+      return events.Sequence([
+          events.RemoveToken(self, "sanity", owner, self.tokens["sanity"]),
+          events.RemoveToken(self, "stamina", owner, self.tokens["stamina"]),
+      ], owner)
+    return None
+
+  def get_usable_interrupt(self, event, owner, state):
+    if (
+        not isinstance(event, events.SliderInput)
+        or event.character != owner
+        or self.get_bonus("abnormal_focus", None, owner, state) >= 3
```

Comment:
Because she may get an abnormal focus bonus from her personal story, I think you want to check the number of tokens on this asset instead of the total bonus.

Reply:
- The Spy's PassStory just lets her use the story instead of spending a sanity or stamina
- The FailStory reduces the number of Abnormal Focus available to the character (probably implemented as 
```python
def get_bonus():
    if check_type == "abnormal_focus":
        return -1
```
- `self.get_bonus` just checks the number of tokens, so works out the same, but saves me from rewriting `sum(self.tokens.values())`.

Reply:
"saves me" as if `sum(self.tokens.values())` is longer than `self.get_bonus("abnormal_focus", None, owner, state) >= 3`. 🙄

Reply:
Ah, I see. I thought it gave her one extra shift, but it needs to be exhausted to give her one extra shift. That is a small, but important distinction that I did not pick up on.